azurerm/resource_arm_key_vault_secret.go

```diff
@@ -131,9 +144,28 @@ func resourceArmKeyVaultSecretCreate(d *schema.ResourceData, meta interface{}) e
 	t := d.Get("tags").(map[string]interface{})
 
 	parameters := keyvault.SecretSetParameters{
-		Value:       utils.String(value),
-		ContentType: utils.String(contentType),
-		Tags:        tags.Expand(t),
+		Value:            utils.String(value),
+		ContentType:      utils.String(contentType),
+		Tags:             tags.Expand(t),
+		SecretAttributes: &keyvault.SecretAttributes{},
+	}
+
+	if v, ok := d.GetOk("not_before_date"); ok {
+		notBeforeDate, err := time.Parse(time.RFC3339, v.(string))
+		if err != nil {
+			return fmt.Errorf("error parsing `not_before_date` time: %s", err)
+		}
```

Comment:
technically we don't need to check for an error here because the validation function should catch all non valid time strings

Reply:
makes sense, removed 👍